knative-operator/pkg/common/metrics.go

```diff
@@ -0,0 +1,26 @@
+package common
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	KnativeServingReadyG = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "knative_serving_ready",
+			Help: "Reports if Knative Serving is up",
+		},
+	)
+	KnativeEventingReadyG = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "knative_eventing_ready",
+			Help: "Reports if Knative Eventing is up",
+		},
```

Comment:
@s-urbaniak these are the metrics.

Reply:
nit: since the help text says "Reports if Knative Eventing is up", why not call it `knative_eventing_up` ?
Additionally you could merge it into one metric called `knative_up{type="eventing"}` and `knative_up{type="serving"}`.

After the naming changes this lgtm as the amount of series is 2. However pinging @lilic and/or @paulfantom for another best-practice pair of eyes.

And pinging @smarterclayton for approval as those seem to be the candidates to be sent via telemetry?

Reply:
> Is it true that only the ones that are part of the cluster-monitoring-operator are pushing data to telemetry?
currently, yes

Reply:
@s-urbaniak thanks for the suggestions I will update as needed. 
Currently there was a request in SRVKE-357 to show the relevant info for eventing to telemeter, so I guess these are going to be the appropriate metrics, still need to discuss internally but wanted to do a dry run as well.

Reply:
Can you explain what are you trying to observe with this metric? Looks like a health metric to me?

Are you installed via OLM, if yes then integrating into cluster monitoring operator should be easy enough.

Reply:
Knowing if it's up or down is helpful, but what action will we take if we observe some are down? Should we be reporting something about why it's down instead of just whether it's up or down? If we ever want to report on specific reasons why it's down, then should we just do that from the outset instead of having an up/down metric that gets obsoleted by one with more information?

Reply:
I would scrape the components not the operator, that way you get the up metric out of the box.

Reply:
@lilic I am not scraping the operator, I am reporting the status of the components which is stated in the operator CR.